### ISSUE TITLE
Retrofit potential tool in master: KeyError: 'Qhsf_MWhyr'

### ISSUE THREAD

**Issue by daren-thomas** [architecture-building-systems/CityEnergyAnalyst#1476]
Running the `cea retrofit-potential` tool on the reference-case-open (`cea extract-reference-case`) produces the following error:

```
C:\Users\darthoma\Documents\GitHub\CityEnergyAnalyst (master)
λ cea retrofit-potential
Running `cea retrofit-potential` with the following parameters:
- general:scenario = C:\reference-case-open\baseline
- retrofit-potential:keep-partial-matches = True
- retrofit-potential:retrofit-scenario-name = retrofit-HVAC
- retrofit-potential:retrofit-target-year = 2020
- retrofit-potential:age-threshold = 15
- retrofit-potential:eui-heating-threshold = 150.0
- retrofit-potential:eui-hot-water-threshold = 50.0
- retrofit-potential:eui-cooling-threshold = 4.0
- retrofit-potential:eui-electricity-threshold = 20.0
- retrofit-potential:emissions-operation-threshold = 30.0
- retrofit-potential:heating-costs-threshold = 2.0
- retrofit-potential:hot-water-costs-threshold = 2.0
- retrofit-potential:cooling-costs-threshold = 2.0
- retrofit-potential:electricity-costs-threshold = 2.0
- retrofit-potential:heating-losses-threshold = 15.0
- retrofit-potential:hot-water-losses-threshold = 15.0
- retrofit-potential:cooling-losses-threshold = 15.0
Running retrofit-potential for scenario = C:\reference-case-open\baseline
Running retrofit-potential with hot-water-costs-threshold = 2.0
Running retrofit-potential with age-threshold = 15
Running retrofit-potential with emissions-operation-threshold = 30.0
Running retrofit-potential with eui-electricity-threshold = 20.0
Running retrofit-potential with heating-costs-threshold = 2.0
Running retrofit-potential with eui-hot-water-threshold = 50.0
Running retrofit-potential with electricity-costs-threshold = 2.0
Running retrofit-potential with heating-losses-threshold = 15.0
Running retrofit-potential with cooling-losses-threshold = 15.0
Running retrofit-potential with keep-partial-matches = True
Running retrofit-potential with retrofit-target-year = 2020
Running retrofit-potential with eui-cooling-threshold = 4.0
Running retrofit-potential with hot-water-losses-threshold = 15.0
Running retrofit-potential with eui-heating-threshold = 150.0
Running retrofit-potential with cooling-costs-threshold = 2.0
Running retrofit-potential with retrofit-scenario-name = retrofit-HVAC
Traceback (most recent call last):
  File "C:\Users\darthoma\AppData\Local\conda\conda\envs\cea\Scripts\cea-script.py", line 11, in <module>
    load_entry_point('cityenergyanalyst', 'console_scripts', 'cea')()
  File "c:\users\darthoma\documents\github\cityenergyanalyst\cea\interfaces\cli\cli.py", line 48, in main
    script_module.main(config)
  File "c:\users\darthoma\documents\github\cityenergyanalyst\cea\analysis\retrofit\retrofit_potential.py", line 238, in main
    emissions_operation_threshold=config.retrofit_potential.emissions_operation_threshold)
  File "c:\users\darthoma\documents\github\cityenergyanalyst\cea\analysis\retrofit\retrofit_potential.py", line 113, in retrofit_main
    ("c_eui_" + criteria_name, eui_filter_HVAC(demand_totals, criteria_name, criteria_threshold)))
  File "c:\users\darthoma\documents\github\cityenergyanalyst\cea\analysis\retrofit\retrofit_potential.py", line 32, in eui_filter_HVAC
    demand["eui"] = demand[variable_name + column_suffix] / demand["Af_m2"] * 1000
  File "C:\Users\darthoma\AppData\Local\conda\conda\envs\cea\lib\site-packages\pandas\core\frame.py", line 1964, in __getitem__
    return self._getitem_column(key)
  File "C:\Users\darthoma\AppData\Local\conda\conda\envs\cea\lib\site-packages\pandas\core\frame.py", line 1971, in _getitem_column
    return self._get_item_cache(key)
  File "C:\Users\darthoma\AppData\Local\conda\conda\envs\cea\lib\site-packages\pandas\core\generic.py", line 1645, in _get_item_cache
    values = self._data.get(item)
  File "C:\Users\darthoma\AppData\Local\conda\conda\envs\cea\lib\site-packages\pandas\core\internals.py", line 3590, in get
    loc = self.items.get_loc(item)
  File "C:\Users\darthoma\AppData\Local\conda\conda\envs\cea\lib\site-packages\pandas\core\indexes\base.py", line 2444, in get_loc
    return self._engine.get_loc(self._maybe_cast_indexer(key))
  File "pandas\_libs\index.pyx", line 132, in pandas._libs.index.IndexEngine.get_loc (pandas\_libs\index.c:5280)
  File "pandas\_libs\index.pyx", line 154, in pandas._libs.index.IndexEngine.get_loc (pandas\_libs\index.c:5126)
  File "pandas\_libs\hashtable_class_helper.pxi", line 1210, in pandas._libs.hashtable.PyObjectHashTable.get_item (pandas\_libs\hashtable.c:20523)
  File "pandas\_libs\hashtable_class_helper.pxi", line 1218, in pandas._libs.hashtable.PyObjectHashTable.get_item (pandas\_libs\hashtable.c:20477)
KeyError: 'Qhsf_MWhyr'
```

@JIMENOFONSECA any idea how to fix this? can you take over this issue please? what is the priority of this bug?

**Comment by JIMENOFONSECA** [comment]
Yes, please assign it to me. Thanks!
On 11 Jun 2018, 20:41 +0800, Daren Thomas <notifications@github.com>, wrote:

Running the cea retrofit-potential tool on the reference-case-open (cea extract-reference-case) produces the following error:

C:\Users\darthoma\Documents\GitHub\CityEnergyAnalyst (master)
λ cea retrofit-potential
Running `cea retrofit-potential` with the following parameters:
- general:scenario = C:\reference-case-open\baseline
- retrofit-potential:keep-partial-matches = True
- retrofit-potential:retrofit-scenario-name = retrofit-HVAC
- retrofit-potential:retrofit-target-year = 2020
- retrofit-potential:age-threshold = 15
- retrofit-potential:eui-heating-threshold = 150.0
- retrofit-potential:eui-hot-water-threshold = 50.0
- retrofit-potential:eui-cooling-threshold = 4.0
- retrofit-potential:eui-electricity-threshold = 20.0
- retrofit-potential:emissions-operation-threshold = 30.0
- retrofit-potential:heating-costs-threshold = 2.0
- retrofit-potential:hot-water-costs-threshold = 2.0
- retrofit-potential:cooling-costs-threshold = 2.0
- retrofit-potential:electricity-costs-threshold = 2.0
- retrofit-potential:heating-losses-threshold = 15.0
- retrofit-potential:hot-water-losses-threshold = 15.0
- retrofit-potential:cooling-losses-threshold = 15.0
Running retrofit-potential for scenario = C:\reference-case-open\baseline
Running retrofit-potential with hot-water-costs-threshold = 2.0
Running retrofit-potential with age-threshold = 15
Running retrofit-potential with emissions-operation-threshold = 30.0
Running retrofit-potential with eui-electricity-threshold = 20.0
Running retrofit-potential with heating-costs-threshold = 2.0
Running retrofit-potential with eui-hot-water-threshold = 50.0
Running retrofit-potential with electricity-costs-threshold = 2.0
Running retrofit-potential with heating-losses-threshold = 15.0
Running retrofit-potential with cooling-losses-threshold = 15.0
Running retrofit-potential with keep-partial-matches = True
Running retrofit-potential with retrofit-target-year = 2020
Running retrofit-potential with eui-cooling-threshold = 4.0
Running retrofit-potential with hot-water-losses-threshold = 15.0
Running retrofit-potential with eui-heating-threshold = 150.0
Running retrofit-potential with cooling-costs-threshold = 2.0
Running retrofit-potential with retrofit-scenario-name = retrofit-HVAC
Traceback (most recent call last):
  File "C:\Users\darthoma\AppData\Local\conda\conda\envs\cea\Scripts\cea-script.py", line 11, in <module>
    load_entry_point('cityenergyanalyst', 'console_scripts', 'cea')()
  File "c:\users\darthoma\documents\github\cityenergyanalyst\cea\interfaces\cli\cli.py", line 48, in main
    script_module.main(config)
  File "c:\users\darthoma\documents\github\cityenergyanalyst\cea\analysis\retrofit\retrofit_potential.py", line 238, in main
    emissions_operation_threshold=config.retrofit_potential.emissions_operation_threshold)
  File "c:\users\darthoma\documents\github\cityenergyanalyst\cea\analysis\retrofit\retrofit_potential.py", line 113, in retrofit_main
    ("c_eui_" + criteria_name, eui_filter_HVAC(demand_totals, criteria_name, criteria_threshold)))
  File "c:\users\darthoma\documents\github\cityenergyanalyst\cea\analysis\retrofit\retrofit_potential.py", line 32, in eui_filter_HVAC
    demand["eui"] = demand[variable_name + column_suffix] / demand["Af_m2"] * 1000
  File "C:\Users\darthoma\AppData\Local\conda\conda\envs\cea\lib\site-packages\pandas\core\frame.py", line 1964, in __getitem__
    return self._getitem_column(key)
  File "C:\Users\darthoma\AppData\Local\conda\conda\envs\cea\lib\site-packages\pandas\core\frame.py", line 1971, in _getitem_column
    return self._get_item_cache(key)
  File "C:\Users\darthoma\AppData\Local\conda\conda\envs\cea\lib\site-packages\pandas\core\generic.py", line 1645, in _get_item_cache
    values = self._data.get(item)
  File "C:\Users\darthoma\AppData\Local\conda\conda\envs\cea\lib\site-packages\pandas\core\internals.py", line 3590, in get
    loc = self.items.get_loc(item)
  File "C:\Users\darthoma\AppData\Local\conda\conda\envs\cea\lib\site-packages\pandas\core\indexes\base.py", line 2444, in get_loc
    return self._engine.get_loc(self._maybe_cast_indexer(key))
  File "pandas\_libs\index.pyx", line 132, in pandas._libs.index.IndexEngine.get_loc (pandas\_libs\index.c:5280)
  File "pandas\_libs\index.pyx", line 154, in pandas._libs.index.IndexEngine.get_loc (pandas\_libs\index.c:5126)
  File "pandas\_libs\hashtable_class_helper.pxi", line 1210, in pandas._libs.hashtable.PyObjectHashTable.get_item (pandas\_libs\hashtable.c:20523)
  File "pandas\_libs\hashtable_class_helper.pxi", line 1218, in pandas._libs.hashtable.PyObjectHashTable.get_item (pandas\_libs\hashtable.c:20477)
KeyError: 'Qhsf_MWhyr'


@JIMENOFONSECA<https://github.com/JIMENOFONSECA> any idea how to fix this? can you take over this issue please? what is the priority of this bug?

**Comment by martin-mosteiro** [comment]
Same issue on my computer.
@daren-thomas I believe the issue is that Qhsf has been renamed to Qhs_sys (see PR #1454 and issue #1455). @JIMENOFONSECA are you working on this?

**Comment by JIMENOFONSECA** [comment]
That is exactly the issue Martin. I am not currently working on this. Would you mind making the changes? I already left the office.
On 12 Jun 2018, 18:29 +0800, Martín Mosteiro-Romero <notifications@github.com>, wrote:

Same issue on my computer.
@daren-thomas<https://github.com/daren-thomas> I believe the issue is that Qhsf has been renamed to Qhs_sys (see PR #1454<https://github.com/architecture-building-systems/CityEnergyAnalyst/pull/1454> and issue #1455<https://github.com/architecture-building-systems/CityEnergyAnalyst/issues/1455>). @JIMENOFONSECA<https://github.com/JIMENOFONSECA> are you working on this?

**Comment by martin-mosteiro** [comment]
@daren-thomas I fixed the KeyError issue shown above in a separate branch. However, once this issue was fixed a new KeyError popped up as the keys in the cost threshold did not match the ones generated in `operation_costs.py`. This now works on my computer.

However, the solution is not as elegant as the code that was there before. So I see two solutions: either change the outputs of the cost calculations (which I'm sure @BhargavaKrishnaSreepathi won't be happy about) or do something like what I did. @JIMENOFONSECA I leave it up to you to decide.

**Comment by daren-thomas** [comment]
@martin-mosteiro which branch do you mean?

**Comment by martin-mosteiro** [comment]
`i1476-Retrofit-potential-tool-KeyError-Qhsf_MWhyr`, which was merged in PR #1496. I requested your review on this PR but I think you did not have a chance to do so.

@daren-thomas Could you test the functionality of the tool on your computer and close the issue if it is solved?
